lib/absinthe/type/built_ins/introspection.ex

```diff
@@ -191,11 +203,24 @@ defmodule Absinthe.Type.BuiltIns.Introspection do
 
     field :input_fields,
       type: list_of(non_null(:__inputvalue)),
+      args: [
+        include_deprecated: [
+          type: :boolean,
+          default_value: false
+        ]
+      ],
       resolve: fn
-        _, %{source: %Absinthe.Type.InputObject{fields: fields}} ->
+        %{include_deprecated: show_deprecated},
+        %{source: %Absinthe.Type.InputObject{fields: fields}} ->
           input_fields =
             fields
-            |> Map.values()
+            |> Enum.flat_map(fn {_, %{deprecation: is_deprecated} = value} ->
+              if !is_deprecated || (is_deprecated && show_deprecated) do
+                [value]
+              else
+                []
+              end
+            end)
```

Comment:
Ah I see this is extracting the values and filtering in one go. I am not entirely sure that this is actually better, I think it'd be clearer as:

```
|> Map.values
|> Enum.filter(fn %{deprecation: is_deprecated} -> !is_deprecated || (is_deprecated && show_deprecated) end)
```

When you `some_map |> Enum.flat_map` Enumerable ends up calling `Enum.to_list` on the map anyway so we may as well do `Map.values |> Enum.filter`.